### PR TITLE
[IDE/CodeCompletion] Register the request evaluator.

### PR DESCRIPTION
**Pull request by dcci**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test

File: lib/IDE/REPLCodeCompletion.cpp [hunk 1]
```diff
@@ -194,6 +194,10 @@ doCodeCompletion(SourceFile &SF, StringRef EnteredCode, unsigned *BufferID,
                  CodeCompletionCallbacksFactory *CompletionCallbacksFactory) {
   // Temporarily disable printing the diagnostics.
   ASTContext &Ctx = SF.getASTContext();
+
+  // Register the request evaluator.
+  registerIDERequestFunctions(Ctx.evaluator);
```
Comment (slavapestov) [inline lib/IDE/REPLCodeCompletion.cpp]: I think you only want to do this once when the ASTContext is being set up, not every time.

**Review by slavapestov — CHANGES_REQUESTED** [review]

File: lib/IDE/REPLCodeCompletion.cpp (hunk 1 above)
Comment (dcci) [reply lib/IDE/REPLCodeCompletion.cpp]: Cool, I'll change that, thanks!

**Comment by nkcsgexi** [comment]
Notice we usually call `registerIDERequestFunctions` after `CompilerInstance::setup(Invocation)` and call it only once in an ASTContext's life cycle. Actually, an assertion will be hit if we call it twice.

File: lib/IDE/REPLCodeCompletion.cpp (hunk 1 above)
Comment (dcci) [reply lib/IDE/REPLCodeCompletion.cpp]: Done.

**Comment by dcci** [comment]
@swift-ci test

**Pull request merged**